File: output/ChatGPT/codenet/C++/Java/s950360470.java
import java.util.Scanner;

public class s950360470 {
    
    static final long MOD = 1000000007;
    
    // a^n
    static long pow(long a, long n) {
        long res = 1;
        while (n > 0) {
            if ((n & 1) == 1) res *= a;
            a *= a;
            n >>= 1;
        }
        return res;
    }
    
    // a^n % mod
    static long modPow(long a, long n, long mod) {
        long res = 1;
        while (n > 0) {
            if ((n & 1) == 1) res = (res * a) % mod;
            a = (a * a) % mod;
            n >>= 1;
        }
        return res;
    }
    
    // a^(-1) % mod (only if : a % p != 0)
    static long modInv(long a, long mod) {
        return modPow(a, mod - 2, mod);
    }
    
    // nCk % mod
    static long modNCk(int n, int k, int mod) {
        if (n == 0 || k == 0) return 1;
        long p = 1, q = 1;
        for (int i = n - k + 1; i <= n; i++)
            p = (p * i) % MOD;
        for (int i = 2; i <= k; i++)
            q = (q * i) % MOD;
        q = modInv(q, MOD);
        return (p * q) % MOD;
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();
        System.out.println(modNCk(k, n, MOD));
    }
}

File: output/ChatGPT/codenet/C++/Python/s950360470.py
MOD = 10**9 + 7

# a^n
def pow(a, n):
    res = 1
    while n > 0:
        if n & 1:
            res *= a
        a *= a
        n >>= 1
    return res

# a^n % mod
def mod_pow(a, n, mod):
    res = 1
    while n > 0:
        if n & 1:
            res = res * a % mod
        a = a * a % mod
        n >>= 1
    return res

# a^(-1) % mod (only if : a % p != 0)
def mod_inv(a, mod):
    return mod_pow(a, mod - 2, mod)

# nCk % mod
def mod_nCk(n, k, mod):
    if n == 0 or k == 0:
        return 1
    p = 1
    q = 1
    for i in range(n - k + 1, n + 1):
        p = (p * i) % MOD
    for i in range(2, k + 1):
        q = (q * i) % MOD
    q = mod_inv(q, MOD)
    return (p * q) % MOD

n, k = map(int, input().split())
print(mod_nCk(k, n, MOD))